### PR TITLE
fix(supervisor): activation logging + skip doomed post-rollback re-gate

### PR DESCRIPTION
**Pull request by pablonyx**
## PR #1223 follow-ups (from the real-process smoke)

Three non-blocking follow-ups surfaced by the post-merge real-process smoke of the supervisor-owned update mechanism (#1223). No correctness change — all deterministic tests still green.

1. **Observability** — emit an `info!` when an update request reaches a terminal result (activated / rolled_back / invalid). Previously only retries logged, so a successful mailbox-triggered activation was silent versus the run-loop's spawn logging.
2. **Efficiency** — after a rollback, wait for plain AnyHarness reachability (`anyharness_healthy(None)`) instead of re-running the version-gated `health_gate` against the *failed* request's version. That re-gate could never pass on the just-restored last-good binary, so it burned a full `health_check_attempts × delay` cycle (~10s in a typical config) for a result that's explicitly discarded (`let _ =`).
3. **Docs** — record in the supervisor structure README (new Operational Notes) that a persistent TLS-trust failure at the artifact CDN buckets as a transient `DownloadTransport` error (retried forever, never terminal `Invalid`) — consistent with R9-002's intent, but worth an ops note; watch heartbeat staleness / desired-vs-observed divergence.

**Proofs:** `cargo test -p proliferate-supervisor` 40 pass; clippy clean; repo-shape + server-boundary checks pass. `supervisor_owned_runtime` stays OFF (flag untouched).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: anyharness/crates/proliferate-supervisor/src/update/activate/mod.rs [hunk 1]
```diff
@@ -287,11 +296,14 @@ async fn roll_back<H: ActivationHost>(
...
-            // reported active and a genuine last-good is now serving.
+            // Best-effort: bring the restored last-good back up and wait for
+            // plain reachability. Deliberately NOT `health_gate(request)` — that
+            // asserts the FAILED request's version, which the restored binary can
+            // never report, so it burned a full attempts×delay cycle for nothing
+            // (smoke follow-up). The outcome is RolledBack — the point is that
+            // the new version is never reported active and last-good is serving.
             let _ = restart_in_order(host, request.component).await;
-            let _ = health_gate(host, request).await;
+            let _ = host.anyharness_healthy(None).await;
```
Comment (pablonyx) [inline anyharness/crates/proliferate-supervisor/src/update/activate/mod.rs]: **SUF-001 — Blocker: preserve the Worker-liveness half of the post-rollback recovery gate.**

Replacing `health_gate(host, request)` is correct because that function compares the restored binary with the failed request's version. But replacing the whole gate with only `anyharness_healthy(None)` also removes `worker_alive()`. For a Worker update, the restored Worker can fail to restart while the unrelated AnyHarness probe succeeds; for an AnyHarness update, the dependent Worker can likewise be dead after the rollback restart.

Please use a version-neutral rollback/recovery gate that requires both plain AnyHarness reachability and Worker liveness, without `worker_reports_version(request.version)`. Add a focused regression test proving the rollback path calls `anyharness_healthy(None)`, still checks Worker liveness, and never rechecks the failed expected version. This stays within the stated efficiency follow-up and preserves the frozen restart/re-gate contract.

**Comment by pablonyx** [comment]
Review at exact head `19246e6a191c7ba84c5c37118b7ffecf7af28172` against current main `83d9d039efd493fb9d32382f76765380693a8212`:

- **SUF-001 — Blocker:** the post-rollback change correctly avoids comparing the restored binary to the failed target version, but it also drops the existing Worker-liveness check. I left the narrow fix and regression-test requirement inline: https://github.com/proliferate-ai/proliferate/pull/1241#discussion_r3590063261
- No other blocking or material findings in this two-file follow-up.
- The failing `intent-tests (provisional)` check is the known unrelated ProductClient workspace-entry failure and should **not** block this PR. Cargo, clippy/test coverage represented by CI, repo-shape, CodeQL, and the remaining checks are green.
- The live E2B N-1→N proof and flag flip remain correctly deferred until managed-cloud qualification lands; this small corrective PR does not need to wait for them.

Fast path: restore a version-neutral recovery gate of plain AnyHarness reachability **plus** Worker liveness, add the focused regression test, rerun the supervisor tests/clippy, and request re-review. This should then be mergeable immediately.

File: anyharness/crates/proliferate-supervisor/src/update/activate/mod.rs (hunk 1 above)
Comment (pablonyx) [reply anyharness/crates/proliferate-supervisor/src/update/activate/mod.rs]: **Fixed at `d076019aa`.**

The post-rollback recovery gate now probes both legs, version-neutral, no short-circuit:

```rust
let _ = restart_in_order(host, request.component).await;
let anyharness_reachable = host.anyharness_healthy(None).await;
let worker_reachable = host.worker_alive().await;
let _ = anyharness_reachable && worker_reachable;
```

- `anyharness_healthy(None)` — plain reachability, no version asserted (the restored binary can't report the failed request's version).
- `worker_alive()` — restored for both the AnyHarness-update case (dependent Worker restarted, can be dead) and the Worker-update case (restored Worker can fail to restart). Evaluated unconditionally so the Worker leg always fires; `worker_reports_version` is deliberately not called here.

Regression `unhealthy_activation_rolls_back_to_last_good` now asserts `worker_alive_calls == 1`. That count can only come from the rollback path: the activation health gate short-circuits on the unhealthy AnyHarness (`anyharness_healthy() && worker_alive()`) before reaching `worker_alive`, so the sole `worker_alive` invocation is the post-rollback probe. Before the fix the count was 0.

Proof at head `d076019aa`: 40/40 supervisor tests pass, clippy clean, `cargo check --workspace` clean, anyharness boundary/old-path/max-lines gates green. No fmt/clippy CI gate exists (verified in `ci.yml`); the pre-existing `rollback.rs` rustfmt diff is on `origin/main` unchanged and not mine.

**Comment by pablonyx** [comment]
## PR_READY

- **Repo / PR:** proliferate-ai/proliferate #1241 — `fix(supervisor): activation logging + skip doomed post-rollback re-gate`
- **Frozen spec:** `~/Vault/Vault/Workspace/Testing/Make Managed Runtime Updates Supervisor-Owned.md` (PR 9 implementation half; this PR is the 3 non-blocking smoke follow-ups from that note's FOLLOW-UPS section)
- **Base SHA:** `83d9d039efd493fb9d32382f76765380693a8212` (origin/main, merge-base clean)
- **Head SHA:** `d076019aa1ff66811aa9f7ab4fb7e5e8a26d5c9f`
- **Session ID:** `6b879480-7f66-461d-91e5-61fe8084e6d2`

### What changed since last review
`SUF-001` (blocker) fixed. The post-rollback recovery gate is version-neutral and probes **both** legs without short-circuit:
- `anyharness_healthy(None)` — plain reachability (no version asserted)
- `worker_alive()` — restored for both AnyHarness-update (dependent Worker) and Worker-update (restored Worker) cases; `worker_reports_version` deliberately not called on the rollback path

Regression added: `unhealthy_activation_rolls_back_to_last_good` asserts `worker_alive_calls == 1`. The activation health gate short-circuits on the unhealthy AnyHarness before reaching `worker_alive`, so the only invocation is the post-rollback probe (was 0 before the fix).

### Proof at head `d076019aa`
- 40/40 `proliferate-supervisor` tests pass
- `cargo clippy -p proliferate-supervisor --all-targets` clean
- `cargo check --workspace` clean (pre-existing anyharness-lib dead-code warnings only)
- Repo gates green: anyharness boundaries, old-paths, max-lines, repo-shape, PR title/labels

### Failures / deviations
- None attributable. `rollback.rs` has a rustfmt diff — it is **pre-existing on origin/main, unchanged by this PR**, and there is **no fmt/clippy CI gate** (verified in `ci.yml`: Rust gates are `cargo check --workspace` + `cargo test --workspace` only).
- `intent-tests (provisional)` / `Shared frontend packages` — known main-wide ProductClient breakage, not PR-attributable.

Deferred (correctly out of scope, gated on PR 2 #1239 + PR 10): live E2B Tier-4 N-1→N proof, `supervisor_owned_runtime` flag flip, `T4-RUNTIME-1` collector.

Not merging — awaiting Pablo's review.

**Comment by pablonyx** [comment]
**E2E CONTROL delta verdict — exact custody**

Base: `83d9d039efd493fb9d32382f76765380693a8212`  
Head: `d076019aa1ff66811aa9f7ab4fb7e5e8a26d5c9f`  
Prior reviewed head: `19246e6a191c7ba84c5c37118b7ffecf7af28172`

**SUF-001 — RESOLVED.** The rollback path now:

1. restarts the restored component(s);
2. awaits `anyharness_healthy(None)`;
3. independently awaits `worker_alive()`; and
4. does not call `worker_reports_version` or compare against the failed request version.

Because the two awaits occur before combining their booleans, a false AnyHarness result cannot short-circuit the Worker probe. The focused regression is meaningful: its activation gate receives an unhealthy AnyHarness and therefore never reaches `worker_alive`; the asserted single Worker probe can only come from the post-rollback path. Exact code inspection also confirms the rollback probe passes `None`, while the original activation gate retains its target-version check.

No new blocker/material findings in the delta. This exact head is **CONTROL_CLEAN / MERGE_READY** on code and contract custody. It remains draft and should not merge until its still-running required checks settle:

- `Cargo check & test`
- `Web frontend build`
- `Shared frontend packages`

`intent-tests (provisional)` is also still running but remains non-blocking/provisional; judge any failure by its actual attribution rather than this Rust-only delta. All other reported checks are green or intentionally skipped. The deferred live E2B N-1→N proof and flag flip remain correctly outside this follow-up.

**Comment by pablonyx** [comment]
## MERGE_READY

- **Repo / PR:** proliferate-ai/proliferate #1241 — `fix(supervisor): activation logging + skip doomed post-rollback re-gate`
- **Frozen spec:** `~/Vault/Vault/Workspace/Testing/Make Managed Runtime Updates Supervisor-Owned.md` (PR 9 smoke follow-ups)
- **Base SHA:** `83d9d039efd493fb9d32382f76765380693a8212` (origin/main, clean merge-base)
- **Head SHA:** `d076019aa1ff66811aa9f7ab4fb7e5e8a26d5c9f`
- **Session ID:** `6b879480-7f66-461d-91e5-61fe8084e6d2`

### Review state
`SUF-001` (only blocker) fixed at `d076019aa`, replied, and thread resolved. No open review threads.

### Contents (three non-blocking smoke follow-ups from the PR 9 note)
1. Info-level log on a terminal mailbox-triggered activation result (observability parity with run-loop spawn logging).
2. Post-rollback recovery gate skips the doomed version-gated re-gate — now version-neutral, probing **both** `anyharness_healthy(None)` and `worker_alive()` (SUF-001 fix).
3. TLS-trust ops note (bounded-retry consideration) in the supervisor README.

### Proof at exact head `d076019aa`
- 40/40 `proliferate-supervisor` tests pass (incl. new SUF-001 regression asserting `worker_alive_calls == 1` on the rollback path)
- `cargo clippy -p proliferate-supervisor --all-targets` clean
- CI green: **Cargo check & test** (8m15s), Analyze (rust/python/js-ts) / CodeQL, Repo shape checks, Validate PR title and labels, Shared frontend packages, Desktop + Web frontend build, Build SDK, CI/CD config checks, Workflow definition lifecycle, intent-billing

### Remaining red (not attributable, not merge-blocking)
- `intent-tests (provisional)` — single fail `T2-WS-2: Add-Repo entry step renders all three options`, a Playwright frontend `locator.click` timeout. This PR is Rust-supervisor-only (+1 supervisor README); it cannot touch that surface. Known main-wide ProductClient breakage; `(provisional)` gate.

### Deferred (correctly out of scope — gated on PR 2 #1239 + PR 10)
Live E2B Tier-4 N-1→N proof, `supervisor_owned_runtime` flag flip, `T4-RUNTIME-1` collector.

**Not merging — awaiting Pablo's review + merge per standing rule.** Ready for agent review at PR #1241 / SHA `d076019aa`.
